Makefile.toml

```
@@ -82,14 +85,62 @@ script = ["""
   ./generate_assets_metadata.sh assets_manifest.json $WASM_VERION $WASM_OUT_DIR_PATH
 """]
 
+[tasks.run-web]
+script = ["""
+    # We don't call build web as we want to be able to not re-build rust/wasm
+    # every time we change dart code, for the same reason we don't run
+    # `flutter pub get`.
+    echo WARNING: 'THIS DOES NOT CALL "build-web" or "flutter pub get"' >&2
+    ${DART_EXAMPLE_WORKSPACE}/flutter_run_web.sh
+"""]
+
 [tasks.build-web]
 # This will run build-local "unnecessarily". This will be fixed in a later PR.
 env = { WASM_VERION="wasm_bindings", WASM_OUT_DIR_PATH="bindings/dart/example/assets/wasm_bindings" }
 run_task = { name = [ "build-dart", "build-wasm", "generate_assets_metadata"] }
 
 [tasks.build-wasm]
+dependencies = ["build-wasm-parallel", "build-wasm-sequential"]
+
+[tasks.build-wasm-parallel]
+condition_script = ["""
+  # We can't use env_false as it requires the variable to be set
+  # and trying to set a default value for it in `tasks.build-wasm`
+  # seems to not work well either.
+  if [ "${DISABLE_WASM_THREADS:-0}" = 0 ]; then
+    exit 0
+  else
+    exit 1
+  fi
+"""]
+script = ["""
+  if ! [ "$(wasm-opt --version | grep -oE "version [0-9]+" | grep -oE "[0-9]+" )" -ge 101 ]; then
+      echo "Make sure wasm-opt is in the PATH and has version >= 101" >&2
+      echo "Download Url: https://github.com/WebAssembly/binaryen/releases/tag/version_101" >&2
+      exit 1
+  fi
+  echo "build parallel wasm"
+  RUSTFLAGS='-C target-feature=+atomics,+bulk-memory,+mutable-globals' \
+    rustup run nightly \
```

this fails to compile with the latest `nightly` version, one recent version which worked on my machine is `2021-08-17`. it shouldn't matter too much which version we choose, but we should pin it to a working version.
```suggestion
    rustup run nightly-2021-08-17 \
```

---

Didn't we decide to use the bootstrap flag on the stable compiler? Otherwise we should use the nightly that correspond to the stable version we are using.

---

Right, I didn't write that down and I forgot over the 5 days skip.

I also haven tried it yet.

---

if the flag works we should try it as well for #199, otherwise we should use `nightly-2021-03-25` as in #199 which corresponds to the nightly of the same date as the sable 1.51 release.

---

Seem to work :+1: 

I will push commits later today.